edi_xml_oca/components/xml_handler.py

```diff
@@ -28,52 +27,60 @@ def __init__(self, work_context):
             if not hasattr(work_context, key):
                 raise AttributeError(f"`{key}` is required for this component!")
 
-        self.schema = xmlschema.XMLSchema(self._get_xsd_schema_path())
+        self.schema_path, self.schema = self._get_xsd_schema()
 
-    def _get_xsd_schema_path(self):
-        """Lookup for XSD schema."""
+    def _get_xsd_schema(self):
+        """Lookup and parse the XSD schema."""
         try:
             mod_name, path = self.work.schema_path.split(":")
         except ValueError as exc:
             raise ValueError("Path must be in the form `module:path`") from exc
-        return modules.get_resource_path(mod_name, path)
 
-    def _parse_xml(self, file_obj, **kw):
+        schema_path = modules.get_resource_path(mod_name, path)
+        if not schema_path:
+            return UserError(f"XSD schema file not found: {self.work.schema_path}")
+
+        with open(schema_path, "r") as schema_file:
+            return schema_path, etree.XMLSchema(etree.parse(schema_file))
+
+    def _xml_string_to_dict(self, xml_string):
         """Read xml_content and return a data dict.
 
-        :param file_obj: file obj of XML file
+        :param xml_string: str of XML file
         """
-        return DotDict(self.schema.to_dict(file_obj, **kw))
+        parsed_dict = xmltodict.parse(xml_string)
+        root_node = next(iter(parsed_dict))
+        return parsed_dict[root_node]
 
-    def parse_xml(self, file_content, **kw):
+    def parse_xml(self, file_content):
```

Comment:
are we sure we don't need these key args

Reply:
It is indeed in the parameters of `xmltodict.parse` https://github.com/martinblech/xmltodict/blob/master/xmltodict.py#L184

Fixed